File: pizza_party/python/pizza_party.py
class PizzaParty(object):
    def __init__(self):
        self.people = int(input("How many people? "))
        self.pizzas = int(input("How many pizzas do you have? "))
        self.slices_per_pizza = int(input("How many slices do the pizzas have? "))
        self.print_stats()

    def total_slices(self, pizzas, slices_per_pizza):
        total_slices_output = pizzas * slices_per_pizza
        return total_slices_output

    def slices_per_person(self, total_slices_output, people):
        slices_per_person_output = total_slices_output / people
        return slices_per_person_output

    def remaining_slices(self, total_slices_output, people):
        total_slices_output = total_slices_output % people
        return total_slices_output

    def print_stats(self):
        print("{} people with {} pizzas.".format(self.people, self.pizzas))

        per_person_slices = self.total_slices(self.pizzas, self.slices_per_pizza)

        print("Each person gets {} pieces of pizza.".format(
            self.slices_per_person(per_person_slices, self.people)))

        print("There are {} slices leftover.".format(
            self.remaining_slices(per_person_slices, self.people)))

party = PizzaParty()

File: pizza_party/ruby/pizza_party.rb
puts "How many people?"
people = gets.chomp

puts "How many pizzas?"
pizzas = gets.chomp

puts "How many slices do the pizzas have?"
slices_per_pizza = gets.chomp

total_slices = pizzas.to_i * slices_per_pizza.to_i
slices_per_person = total_slices.to_i / people.to_i
remaining_slices = total_slices.to_i % people.to_i

puts "#{people} with #{pizzas}"
puts "Each person gets #{slices_per_person} pieces of pizza."
puts "There are #{remaining_slices} leftover."
